File: backend/src/main/java/com/drivehub/vehiclerental/service/PaymentService.java
package com.drivehub.vehiclerental.service;

import com.drivehub.vehiclerental.dto.PaymentRequest;

/**
 * Service interface for payment operations
 */
public interface PaymentService {
    
    /**
     * Process payment using strategy pattern
     * 
     * @param paymentRequest Payment request
     * @return Payment confirmation message
     */
    String processPayment(PaymentRequest paymentRequest);
}

File: frontend/src/services/paymentService.js
import axiosInstance from './axiosInstance';

// Payment Service
export const paymentService = {
  // Process payment
  processPayment: async (paymentData) => {
    const response = await axiosInstance.post('/payments/process', paymentData);
    return response.data;
  },
};

export default paymentService;
